src/host/inputBuffer.cpp

```diff
@@ -750,23 +751,39 @@ void InputBuffer::_WriteBuffer(_Inout_ std::deque<std::unique_ptr<IInputEvent>>&
                                _Out_ size_t& eventsWritten,
                                _Out_ bool& setWaitEvent)
 {
+    auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+
     eventsWritten = 0;
     setWaitEvent = false;
     const auto initiallyEmptyQueue = _storage.empty();
     const auto initialInEventsSize = inEvents.size();
     const auto vtInputMode = IsInVirtualTerminalInputMode();
 
-    while (!inEvents.empty())
+    for (auto& inEvent : inEvents)
     {
-        // Pop the next event.
+        if (inEvent->EventType() == InputEventType::KeyEvent && static_cast<const KeyEvent*>(inEvent.get())->IsKeyDown())
+        {
+            // if output is suspended, any keyboard input releases it.
+            if (WI_IsFlagSet(gci.Flags, CONSOLE_SUSPENDED) && !IsSystemKey(static_cast<const KeyEvent*>(inEvent.get())->GetVirtualKeyCode()))
+            {
+                UnblockWriteConsole(CONSOLE_OUTPUT_SUSPENDED);
+                continue;
+            }
+            // intercept control-s

```

Comment:
hey uh, does this fix #809?

Reply:
Does <kbd>Ctrl+Q</kbd> un-suspend it?

Reply:
wait, ^S is not VK_PAUSe...

this is the problem with the old comments you found...